src/Functions/ztest.cpp

```diff
@@ -0,0 +1,199 @@
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+        static DataTypePtr getReturnType()
+        {
+            DataTypes types
+            {
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+            };
+
+            Strings names
+            {
+                "z_statistic",
+                "p_value",
+                "confidence_interval_low",
+                "confidence_interval_high"
+            };
+
+            return std::make_shared<DataTypeTuple>(
+                std::move(types),
+                std::move(names)
+            );
+        }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+
+            for (size_t i = 0; i < 4; ++i)
+            {
+                if (!isNativeInteger(arguments[i]))
+                {
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "The {}th Argument of function {} must be a integer or unsigned integer.", i + 1, getName());
+                }
+            }
+
+            if (!isFloat(arguments[4]))
+            {
+                throw Exception{"The fifth argument of function " + getName() + " should be a float, illegal type: " + arguments[4]->getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+            }
+
+            if (!isString(arguments[5]))
+            {
+                throw Exception{"The sixth argument of function " + getName() + " should be a string, illegal type: " + arguments[5]->getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+            }
+
+            return getReturnType();
+        }
+
+
+        static void nan(MutableColumnPtr & to)
+        {
+            const Float64 nan = std::numeric_limits<Float64>::quiet_NaN();
+            Tuple tuple({nan, nan, nan, nan});
+            to->insert(tuple);
+        }
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+        {
+            auto non_const_arguments = arguments;
+            for (auto & argument : non_const_arguments)
+                argument.column = argument.column->convertToFullColumnIfConst();
+
+            MutableColumnPtr to{getReturnType()->createColumn()};
+            to->reserve(input_rows_count);
+
+            if (!isString(arguments[5].type))
+                throw Exception{"The sixth argument of function " + getName() + " should be String, illegal type: " + arguments[5].type->getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+            for (size_t row_num = 0; row_num < input_rows_count; ++row_num)
+            {
+                const UInt64 successes_x = arguments[0].column->getUInt(row_num);
```

Comment:
We assume that in first 4 columns there will be some native integer. Currently we cannot make a lot of template specializations because it will be too much.
First problem can be if value in column is signed and negative, not sure if this makes sense for this statistical function if not then we should add additional check and pass only signed integers in validation.
Second if only signed integers are allowed here, we can unwrap column data and move it into array of UInt64, that way in this hot loop we will work with this arrays, instead of indirection calls to columns.